testing/dart/image_filter_test.dart

```diff
@@ -78,12 +132,19 @@ void main() {
     }
   }
 
+  List<ImageFilter> composed(List<ImageFilter> a, List<ImageFilter> b) {
+    return <ImageFilter>[for (final ImageFilter x in a) for (final ImageFilter y in b) ImageFilter.compose(outer: x, inner: y)];
+  }
+
   test('ImageFilter - equals', () async {
     final List<ImageFilter> A = makeList();
     final List<ImageFilter> B = makeList();
     checkEquality(A, A);
     checkEquality(A, B);
     checkEquality(B, B);
+    checkEquality(composed(A, A), composed(A, A));
+    checkEquality(composed(A, B), composed(B, A));
```

Comment:
Does this pass? They are composed in opposite order so they shouldn't be equal each other. The equality test expects equality when the index of one list matches the index in the other, but the first list will have `outer: a, inner: b` and the second list will have `outer: b, inner: a`...? If that passes then there might be something wrong with the == operators.

Reply:
It does pass I think. `A` and `B` contain the same filters (in terms of `==`) in the exact same order so the output of `composed(A, B)` `composed(B, A)` contain the same filters in the exact same order too.

Reply:
(Smacks forehead). Oh right, so for each pair it would be `compose(A, dup of A) == compose(dup of A, A)` which should be equal.